site/content/extend/plugins/webapp/best-practices.md

```diff
@@ -0,0 +1,67 @@
+- Recommendation: Have your plugin register a [post dropdown menu component](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerPostDropdownMenuComponent) with some text, icon and an action function. This adds your action to the "More Actions" post menu dropdown for easy discovery.
+- Examples: Create or attach to Jira issue from a message; copy a message to another channel; report an inappropriate message
+
+![Post Dropdown Menu](/img/extend/bp-post-dropdown-menu.png)
+
+### Actions related to files or images
+- Recommendation: Have your plugin register a [file upload method](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerFileUploadMethod) with some text, icon and an action function. This adds your new action to the file upload menu.
+- Examples: File sharing from OneDrive or GDrive; Draw plugin for sketches
+
+![File Upload Action](/img/extend/bp-file-upload.png)
+
+
+### Actions that apply to specific teams
+- Recommendation: Have your plugin register [left sidebar header component](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerLeftSidebarHeaderComponent) with some text, icon and an action function. This adds your action above your team's channels in the sidebar.
+- Examples: Trello kanban board plugin, Github Plugin
+
+![post dropdown menu](/img/extend/bp-left-sidebar-header.png)
+
+### Quick links or status summaries of workflows
+- Recommendation: Have your plugin register a [bottom team sidebar component](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerBottomTeamSidebarComponent). This adds icons to the lower left corner of the UI.
+- Examples: GitHub sidebar links with summary of outstanding reviews or unread messages; ServiceNow incident status summary
+
+![post dropdown menu](/img/extend/bp-bottom-team-sidebar.png)
+
+### Global actions that can be taken anywhere in the server and not directly related to teams, channels or users
+- Recommendation: Have your plugin register [main menu action](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerMainMenuAction) with some text, icon for mobile, and an action function. This adds your action to the Main Menu. You can additionally [register a slash command](https://developers.mattermost.com/extend/plugins/server/reference/#API.RegisterCommand) on the server-side.
+- Examples: Share feedback plugin in Main Menu; /jira slash commands for quick actions
+
+![post dropdown menu](/img/extend/bp-main-menu-action.png)
+
+### Actions that apply to specific users
+- Recommendation: Have your plugin register a [popover user actions component](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerPopoverUserActionsComponent). This adds your action button to the user profile popover.   
+- Examples: ..
+
+![popover user actions component](/img/extend/bp-user-popover.png)
+
+### Display custom information on a user profile
```

Comment:
I realized this is the only one that doesn't start with a noun ('actions', 'links'). 0/5 if we want to reword to start with a noun for consistency, e.g. `Custom information on a user profile`